File: leetcode/0461_hamming_distance.kt
package leetcode

class Solution {
    fun hammingDistance(x: Int, y: Int): Int {
        val xs = x.toString(2).padStart(32, '0')
        val ys = y.toString(2).padStart(32, '0')

        return (xs.zip(ys) { xc, yc -> xc != yc }).count({ zb -> zb })
    }
}

fun main(args: Array<String>) {
    val solution = Solution()

    assert(2 == solution.hammingDistance(1, 4))
}

File: leetcode/0461_hamming_distance.py
# -*- coding: utf-8 -*-


class Solution:
    def hammingDistance(self, x, y):
        xs = format(x, "032b")
        ys = format(y, "032b")

        return sum(xc != yc for xc, yc in zip(xs, ys))


if __name__ == "__main__":
    solution = Solution()

    assert 2 == solution.hammingDistance(1, 4)
